modules/beta-private-cluster/scripts/wait-for-cluster.sh

```diff
@@ -25,12 +25,12 @@ CLUSTER_NAME=$2
 
 echo "Waiting for cluster $CLUSTER_NAME in project $PROJECT to reconcile..."
 
-current_status=$(gcloud container clusters list --project="$PROJECT" --filter=name:"$CLUSTER_NAME" --format="value(status)")
+current_status=$(gcloud container clusters list --project="$PROJECT" --filter=name:"$CLUSTER_NAME" --format="value(status)") --impersonate_service_account="$impersonate_service_account"
```

Comment:
Also wouldn't we need to handle the case that impersonate_service_account is empty?

Reply:
I believe this would work as expected with empty impersonate_service_account - no impersonation for the command. 

This can be tested with any gcloud command, such as _gcloud projects list --impersonate_service_account=""_